docs/rackhd/microkernel.rst

```diff
@@ -64,39 +64,72 @@ with the base image built by the existing ansible roles in this repository.
+- **/tmp/on-imagebuilder/ipxe/**      :  it includes the artifacts:
+  * monorail-efi32-snponly.efi
+  * monorail-efi64-snponly.efi
+  * monorail.ipxe
+  * monorail-undionly.kpxe
 
-   **NOTE:** The config_file must have as a bare minimum a provisioner variable that points to the role::
+- **syslinux**/undionly.kkpxe
 
-                      provisioner: roles/overlay/provision_discovery_overlay
 
+- Note: OEM role provision_raid_overlay requires storcli_1.17.08_all.deb being copied into
+  common/files. User can download it from http://docs.avagotech.com/docs/1.17.08_StorCLI.zip.
 
-4. Configure a new playbook (see example.yml) to run the appropriate wrapper playbook with the config_file, for example::
 
-              - include: common/overlay_wrapper.yml
-                 vars:
-                   - config_file: vars/my_overlay.yml
-
-
-The wrapper playbooks handle all the setup and cleanup required to run a provisioner, such as filesystem mounting and creation, and build file creation.
-
-.. _ansible modules: http://docs.ansible.com/ansible/modules_intro.html
+Adding Provisioner Roles and Configuration Files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+The provisioner role is what specifies how the filesystem of an initrd, base
+image or overlay should be customized. To add a new provisioner, do the following.
+If you have experience with Ansible, some of these steps will be familiar:
+
+- Make a new directory in `roles/<initrd|basefs|overlay>/tasks`, depending on the image type
+- Create and edit a main.yml file in the above directory to do the tasks you
+  want (see [ansible modules](http://docs.ansible.com/ansible/modules_intro.html)
+  if new to Ansible)
+- Add a new config yaml file into the vars directory. This will be included in the
+  Ansible run as a set of top-level variables (via the -e argument) to be included/used
+  by tasks in the role.
+- Configure a new script (see example.sh) to run the appropriate wrapper
+  playbook with the config file and the provisioner role specified as vars, for example:
+
+    ```
+    sudo ansible-playbook \
+    -i hosts common/overlay_wrapper.yml \
+    -e "config_file=vars/overlay.yml \
+    provisioner=roles/overlay/provision_discovery_overlay"
+    ```
+  The wrapper playbooks handle all the setup and cleanup required to run a
+  provisioner, such as filesystem mounting and creation, and build file creation.
+- **NOTE**: the customized overlay depends on initrd and basefs build result, that said, before running the overlay playbook, below playbook should run firstly:
+
+    ```
+    sudo ansible-playbook -i hosts common/initrd_wrapper.yml \
+    -e "config_file=vars/initrd.yml provisioner=roles/initrd/provision_initrd"
+
+    sudo ansible-playbook -i hosts common/basefs_wrapper.yml \
+    -e "config_file=vars/basefs.yml provisioner=roles/basefs/provision_rootfs"
+    ```
+
+  so it should run `build_all.sh` which include inirtd and basefs playbook before `build_oem.sh`
 
```

Comment:
Just a small typo to fix the spelling of "initrd".  Also, not sure if this is the right place, but did you want/need any description of commands to run the "build_oem.sh" script?